mmv1/templates/terraform/constants/notebooks_instance.go

```diff
@@ -0,0 +1,28 @@
+var NotebooksInstanceProvidedScopes = []string{
+	"https://www.googleapis.com/auth/cloud-platform",
+	"https://www.googleapis.com/auth/userinfo.email",
+}
+
+func NotebooksInstanceScopesDiffSuppress(_, _, _ string, d *schema.ResourceData) bool {
```

Comment:
I don't fully understand the problem with scopes. On the linked issue it was [noted that the API doesn't return service account scopes](https://github.com/hashicorp/terraform-provider-google/issues/16587#issuecomment-1822871108), but in the debug logs for [this failing test](https://github.com/GoogleCloudPlatform/magic-modules/pull/9915#issuecomment-1927798059) I can see `serviceAccountScopes` is present in responses from the API. I saw that there are two scopes that are [automatically used](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/notebooks_instance#service_account_scopes) when scopes aren't specified in the config by a user, so I can see why there may be a need for a diff suppress.

Could you please explain what scenarios the diff suppress would be needed in?

Reply:
The issue with scopes was that it was not returned in the response. We changed that in the backend to return the scopes. The issue currently is that `default_from_api: true` was not enabled and the 2 default scopes need to be always added for the instance to work, which is why this diff suppress.